rage/vector.hpp

```diff
@@ -27,22 +27,133 @@ namespace rage
 		T data[3];
```

Comment:
Using a union with templating, according to the C++ standard, leads to undefined behavior. This was like this before, but again, if we're making it "better", this should be addressed.